disttask/framework/scheduler/scheduler.go

```diff
@@ -165,14 +166,25 @@ func (s *BaseScheduler) run(ctx context.Context, task *proto.Task) error {
 		wg.Wait()
 	}()
 
+	subtasks, err := s.taskTable.GetSubtasksInStates(s.id, task.ID, task.Step, proto.TaskStatePending)
+	if err != nil {
+		s.onError(err)
+		return s.getError()
+	}
+	for _, subtask := range subtasks {
+		metrics.IncDistTaskSubTaskCnt(subtask)
+		metrics.StartDistTaskSubTask(subtask)
+	}
```

Comment:
We can move this code into dispatcher.go. 
When dispatching subtasks success, update the metric.
Then we don't need to fetch the taskTable.

Reply:
This was my previous implementation method, which would cause the instance of collecting metrics to be different, thereby causing confusion in Grafana display.